tests/unit/test_utils.py

```diff
@@ -66,3 +71,33 @@ def test_double_quotes_inside():
 
 VALUES ({TEST_VALUE_ESCAPED}, 'c')"""
     ), test_insert_query
+
+
+def test_check_if_empty_file(caplog):
+    with open("empty.csv", "w"):
+        pass
+
+    with caplog.at_level(logging.WARNING):
+        check_if_empty_file(path="empty.csv", if_empty="warn", file_extension=".csv")
+        assert "The input file is empty." in caplog.text
+    with pytest.raises(ValueError):
+        check_if_empty_file(path="empty.csv", if_empty="fail", file_extension=".csv")
+    with pytest.raises(SKIP):
+        check_if_empty_file(path="empty.csv", if_empty="skip", file_extension=".csv")
+
+    os.remove("empty.csv")
```

Comment:
You have three different handles inside `check_if_empty_file` I think (CSV, Parquet, and other), so I think this test should handle all three cases (or you can create 3 separate tests if that's easier).